src/game.cpp

```diff
@@ -4104,12 +4104,26 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats, f32 dtime,
 		Update clouds
 	*/
 	if (clouds) {
-		v3f player_position = player->getPosition();
 		if (sky->getCloudsVisible()) {
 			clouds->setVisible(true);
 			clouds->step(dtime);
-			clouds->update(v2f(player_position.X, player_position.Z),
-				       sky->getCloudColor());
+			// camera->getPosition is not enough for 3rd person views
+			clouds->update(camera->getCameraNode()->getPosition(),
+					sky->getCloudColor());
+			if (clouds->isCameraInsideCloud() && m_cache_enable_fog &&
+					!flags.force_fog_off) {
+				// if inside clouds, and fog enabled, use that as sky
+				// color(s)
+				sky->overrideColors(clouds->getColor().getInterpolated(
+								    video::SColor(255, 0,
+										    0, 0),
+								    0.9),
+						clouds->getColor());
+				sky->setBodiesVisible(false);
+				runData.fog_range = 20.0f * BS;
+				clouds->setVisible(
```

Comment:
indent is very... unreadable here, can you improve it to inline some functions properlys?